entry/src/block_component.rs

```
@@ -618,6 +618,49 @@ impl BlockComponent {
             }
         }
     }
+
+    /// Parse BlockHeader from bytes
+    pub fn parse_block_header_from_data_payload(data: &[u8]) -> Option<(Slot, Hash)> {
+        // Try to deserialize as BlockComponent
+        let (component, _) = BlockComponent::from_bytes(data).ok()?;
```

Is it okay to discard all errors silently here? Should we log or add to stats?

---

do metrics get logged on dead slots?

separately - rather than here, it's probably worth adding metrics to `BlockComponentProcessor`; can do this in a follow-up PR